frontend/views/containers/modals/LeaveGroup.vue

```diff
@@ -0,0 +1,158 @@
+            name='name'
+            v-model='form.name'
+            @keyup.enter='submit'
+            @input='$v.form.name.$touch()'
+            placeholder='username'
+            ref='username'
+            autofocus
+            data-test='loginName'
+          )
+          i18n.error(
+            tag='p'
+            v-show='$v.form.name.$error'
+          ) username cannot contain spaces
+
+        form-password(
+          :label='L("Password")'
+          :value='form'
+          :v='$v.form'
+          @enter='submit'
+          @input='(Password) => {password = Password}'
+          data-test='loginPassword'
+        )
+
+        .field
+          i18n.label(tag='label') Type "Leave The Dreamers" below
+
+          input.input(
+            :class="{'error': $v.form.confirmation.$error}"
+            name='confirmation'
+            v-model='form.confirmation'
+            @keyup.enter='submit'
+            @input='$v.form.confirmation.$touch()'
+            placeholder=''
+            data-test='confirmation'
+          )
+
+          p.error(
+            v-show='$v.form.confirmation.$error'
+            v-html="L('Please enter the sentence \"<b>Leave The Dreamers</b>\" to confirm that you leave the group')"
+          )
+
+        .buttons
+          i18n.is-outlined(
+            tag='button'
+            @click='close'
+          ) Cancel
+          i18n.is-danger(
+            tag='button'
+            @click='submit'
+            :disabled='$v.form.$invalid'
+          ) Leave Group
+
+      template(slot='errors') {{ form.response }}
+</template>
+
+<script>
+import { validationMixin } from 'vuelidate'
+import ModalTemplate from '@components/Modal/ModalTemplate.vue'
+import FormPassword from '@components/Forms/Password.vue'
+import { required } from 'vuelidate/lib/validators'
+import { CLOSE_MODAL } from '@utils/events.js'
+import sbp from '~/shared/sbp.js'
+import Message from '@components/Message.vue'
+
+export default {
+  name: 'LeaveGroupModal',
+  mixins: [validationMixin],
+  data () {
+    return {
+      form: {
+        name: null,
+        password: null,
+        confirmation: null
+      }
+    }
+  },
+  validations: {
+    form: {
+      name: {
+        required
+      },
+      password: {
+        required
+      },
+      confirmation: {
+        required,
+        checkConfirmation: value => {
+          console.log(value)
+          return value === 'Leave The Dreamers'
+        }
+      }
+    }
+  },
+  components: {
+    ModalTemplate,
+    FormPassword,
+    Message
+  },
+  methods: {
+    close () {
+      sbp('okTurtles.events/emit', CLOSE_MODAL)
+    },
+    async submit () {
+      console.error('TODO: implement')
+      this.close()
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.modal-card-body {
+  padding-top: 0;
+}
+
+.message.is-danger {
+  width: 100%;
+  padding: 1rem;
+  margin-top: 1rem;
+  margin-bottom: 2rem;
+  align-items: center;
+
+  ::v-deep i {
```

Comment:
In this specific case why do we need the deep selector? Since it's an html tag it works with or without deep, correct?

Reply:
Maybe. Have you tried before this comment?

Reply:
yes, it works without deep.

Reply:
I hate any deep selector. In general if they are there it's for a reason. But if that works, that should be definitely removed.

Reply:
I hate any deep selector. In general if they are there it's for a reason. But if that works, that should be definitely removed.